### PR TITLE
Extend exception messages

### PR DESCRIPTION
**Pull request by orozcodiaz**
This improves the exception message to provide more information.

### CODE REVIEW

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @oroskodias,

Sorry for long delay. Your changes looks good to me. Could you cover your changes with unit or integration test?

Thank you!

**Comment by engcom-Echo** [comment]
I will take care of test coverage.

**Comment by ihor-sviziev** [comment]
Hi @engcom-Echo,
Could you fix author name & email for commit you've pushed & force push?
If you have any questions - feel free to find me in Slack

File: app/code/Magento/Store/Test/Unit/Controller/Store/RedirectTest.php [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+            ->with('___from_store')
+            ->willReturn($defaultStoreViewCode);
+        $this->storeRepositoryMock
+            ->expects($this->once())
+            ->method('get')
+            ->with($defaultStoreViewCode)
+            ->willThrowException(new NoSuchEntityException());
+        $this->messageManagerMock
+            ->expects($this->once())
+            ->method('addErrorMessage')
```
Comment (ihor-sviziev) [inline app/code/Magento/Store/Test/Unit/Controller/Store/RedirectTest.php:247]: @engcom-Echo 
I think purpose of covering this case with unit test - compare if we have correct message, I can't find where is it checked. Could you point me or adjust the test?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Store/Test/Unit/Controller/Store/RedirectTest.php (hunk 1 above)
Comment (engcom-Echo) [reply app/code/Magento/Store/Test/Unit/Controller/Store/RedirectTest.php:247]: Thank you.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6653](https://jira.corp.magento.com/browse/ENGCOM-6653) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Before:**
before

**After:**
after

File: app/code/Magento/Store/Test/Unit/Controller/Store/RedirectTest.php [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+            ->willReturnSelf();
+        $this->storeRepositoryMock
+            ->expects($this->never())
+            ->method('get');
+
+        $this->assertEquals($this->responseMock, $this->redirectController->execute());
+    }
+
+    /**
+     * @inheritDoc
```
Comment (Nazar65) [inline app/code/Magento/Store/Test/Unit/Controller/Store/RedirectTest.php]: Inheritance from what ?

File: app/code/Magento/StoreGraphQl/Test/Unit/StoreValidatorTest.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        $this->storeManagerMock
+             ->expects($this->never())
+             ->method('setCurrentStore')
+             ->with(null)
+             ->willReturnSelf();
+        $this->storeValidator->validate($this->requestMock);
+    }
+
+    /**
+     * @inheritDoc
```
Comment (Nazar65) [inline app/code/Magento/StoreGraphQl/Test/Unit/StoreValidatorTest.php]: Inheritance from what ?

**Review by Nazar65 — CHANGES_REQUESTED** [review]

**Review by Nazar65 — APPROVED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6653](https://jira.corp.magento.com/browse/ENGCOM-6653) has been created to process this Pull Request

**Pull request merged**